### PR TITLE
add vue oauth2 keycloak module

### LINKED ISSUE #1133: Vue: add OAuth2 keycloak authentication
**Prerequisites**
- [x] https://github.com/jhipster/jhipster-lite/pull/10712
- [x] https://github.com/jhipster/jhipster-lite/issues/10876

**TO-DO**
- [x] Create a sample project that adds the OAuth2 Keycloak module to the Vue frontend client and shows it working with login and logout
- [x] Create the `vue-oauth2-keycloak` module in the JHLite project which depends on `vue-core` module
  - [x] Use `keycloak-js` dependency from COMMON package at `AngularOAuth2ModulesFactory` and remove it from ANGULAR package

### PR DESCRIPTION
**Pull request by renanfranca**
Fix #1133 

image

**TO-DO:**
- [x] Add to `tests-ci/generate.sh` this new module at `Vue app`
  - [x] Add unit tests

- [x] Add keycloak-js dependency to common package.json
  - Added "keycloak-js": "25.0.5" to the dependencies in the common package.json file

- [x] Add tests for the VueOAuth2KeycloakModulesFactory
  - Created VueOAuth2KeycloakModulesFactoryTest.java
  - Implemented tests to verify the correct building of the Vue OAuth2 Keycloak module
  - Checked for proper file creation and content in the generated module

- [x] Add template files to be used by vue-oauth2-keycloak module
  - Created AuthProvider.ts.mustache
  - Created AuthRepository.ts.mustache
  - Created AuthenticatedUser.ts.mustache
  - Created KeycloakAuthRepository.ts.mustache
  - Created KeycloakHttp.ts.mustache
  - These files provide the core functionality for OAuth2 authentication with Keycloak in a Vue application

- [x] Remove unused import added to main.ts
  - Removed the unused 'axios' import from the main.ts file in VueOAuth2KeycloakModulesFactory.java

- [x] Add package-info.java
  - Added package-info.java to the vue.security.oauth2_keycloak package
  - Applied @tech.jhipster.lite.BusinessContext annotation to the package

- [x] Add vue-oauth2-keycloak module configuration
  - Created VueOAuth2KeycloakApplicationService.java
  - Implemented VueOAuth2KeycloakModuleConfiguration.java
  - Added VUE_OAUTH2_KEYCLOAK to JHLiteModuleSlug enum

- [x] Add cucumber test to vue-oauth2-keycloak module configuration
  - Created vue-oauth2-keycloak.feature file
  - Implemented scenario to test the application of Vue OAuth2 Keycloak module

- [x] Fix cucumber test
  - Updated vue-oauth2-keycloak.feature
  - Added 'typescript' module to the list of applied modules in the test scenario

- [x] Improve VueOAuth2KeycloakModulesFactory code by using constants
  - Refactored VueOAuth2KeycloakModulesFactory.java
  - Introduced constants for repeated string literals
  - Improved code readability and maintainability

- [x] Use keycloak-js dependency from COMMON package instead of the ANGULAR package
  - Updated AngularOauth2ModuleFactory.java
  - Changed the source of keycloak-js dependency from ANGULAR to COMMON
  - Removed keycloak-js entry from angular/package.json

- [x] Add vue-authentication-components.md documentation to vue-oauth2-keycloak module
  - Created vue-authentication-components.md
  - Provided detailed documentation on the Vue authentication components
  - Included file structure, detailed file explanations, and usage instructions

- [x] Fix sonar issues at KeycloakHttp.ts.mustache
  - Updated KeycloakHttp.ts.mustache
  - Replaced `||` with `??` for null coalescing
  - Improved code quality and addressed potential issues flagged by SonarQube

### CODE REVIEW

File: src/main/resources/generator/client/vue/webapp/app/auth/domain/AuthenticatedUser.ts.mustache [hunk 1]
```diff
@@ -0,0 +1,8 @@
+type AuthenticatedUserName = string;
+type AuthenticatedUserToken = string;
+
+export interface AuthenticatedUser {
```
Comment (DamnClin) [inline src/main/resources/generator/client/vue/webapp/app/auth/domain/AuthenticatedUser.ts.mustache:4]: I think this is a type, not an interface

File: src/main/resources/generator/client/vue/webapp/app/auth/infrastructure/secondary/KeycloakAuthRepository.ts.mustache [hunk 2]
```diff
@@ -0,0 +1,53 @@
+import type { AuthRepository } from '@/auth/domain/AuthRepository';
+import type { AuthenticatedUser } from '@/auth/domain/AuthenticatedUser';
+import { KeycloakHttp } from './KeycloakHttp';
+
+export class KeycloakAuthRepository implements AuthRepository {
```
Comment (DamnClin) [inline src/main/resources/generator/client/vue/webapp/app/auth/infrastructure/secondary/KeycloakAuthRepository.ts.mustache:5]: I think we should stick to promise for error handling and remove console.log

File: src/main/resources/generator/client/vue/webapp/app/auth/domain/AuthRepository.ts.mustache [hunk 3]
```diff
@@ -0,0 +1,9 @@
+import type {AuthenticatedUser} from "@/auth/domain/AuthenticatedUser";
```
Comment (qmonmert) [inline src/main/resources/generator/client/vue/webapp/app/auth/domain/AuthRepository.ts.mustache:1]:
```suggestion
import type { AuthenticatedUser } from "@/auth/domain/AuthenticatedUser";
```

File: src/main/resources/generator/client/vue/webapp/app/auth/infrastructure/secondary/KeycloakHttp.ts.mustache [hunk 4]
```diff
@@ -0,0 +1,49 @@
...
+    await this.ensureInitialized();
+    if (this.keycloak.authenticated) {
+      return {
+        isAuthenticated: true,
+        username: this.keycloak.tokenParsed?.preferred_username ?? '',
+        token: this.keycloak.token ?? '',
+      };
+    } else {
+      return { isAuthenticated: false, username: '', token: '' };
+    }
```
Comment (qmonmert) [inline src/main/resources/generator/client/vue/webapp/app/auth/infrastructure/secondary/KeycloakHttp.ts.mustache:26]:
```suggestion
    if (this.keycloak.authenticated) {
      return {
        isAuthenticated: true,
        username: this.keycloak.tokenParsed?.preferred_username ?? '',
        token: this.keycloak.token ?? '',
      };
    }
    return { isAuthenticated: false, username: '', token: '' };
```

File: src/main/resources/generator/client/vue/documentation/vue-authentication-components.md [hunk 5]
```diff
@@ -0,0 +1,670 @@
+# Vue Authentication Components Documentation
+
+This document provides an overview and showcase the practical usage of the vue authentication module.
```
[the remark this answers is not in the document]
Comment (renanfranca) [reply src/main/resources/generator/client/vue/documentation/vue-authentication-components.md]: Thanks! I fixed it 👍

File: src/main/resources/generator/client/vue/webapp/app/auth/domain/AuthenticatedUser.ts.mustache (hunk 1 above)
Comment (renanfranca) [reply src/main/resources/generator/client/vue/webapp/app/auth/domain/AuthenticatedUser.ts.mustache:4]: I follow the jhipster-lite code:
```typescript
import { ModuleParameterType } from './ModuleParameters';
import { ModulePropertyKey } from './ModulePropertyKey';

export interface ModuleParameter {
  key: ModulePropertyKey;
  value: ModuleParameterType;
}
```

If I add type I get the error:
image

Comment (DamnClin) [reply src/main/resources/generator/client/vue/webapp/app/auth/domain/AuthenticatedUser.ts.mustache:4]: Yes, you need a = sign

Comment (renanfranca) [reply src/main/resources/generator/client/vue/webapp/app/auth/domain/AuthenticatedUser.ts.mustache]: Thank you! I did it 👍

File: src/main/resources/generator/client/vue/webapp/app/auth/domain/AuthRepository.ts.mustache (hunk 3 above)
Comment (renanfranca) [reply src/main/resources/generator/client/vue/webapp/app/auth/domain/AuthRepository.ts.mustache]: Thanks! Fixed 👍

File: src/main/resources/generator/client/vue/webapp/app/auth/infrastructure/secondary/KeycloakAuthRepository.ts.mustache (hunk 2 above)
Comment (renanfranca) [reply src/main/resources/generator/client/vue/webapp/app/auth/infrastructure/secondary/KeycloakAuthRepository.ts.mustache:5]: Thanks ;)! I fixed it 👍

**Review by qmonmert — DISMISSED** [review]

**Comment by renanfranca** [comment]
As I mentioned here:
- https://github.com/jhipster/jhipster-lite/discussions/10927#discussioncomment-10721027

I added the unit tests to ensure this module can be added to the CI pipeline while maintaining the current 100% coverage metric. Once the discussion reaches a consensus, I’ll proceed with refactoring this module accordingly.

I am doing this because I want to conclude this issue before moving on to another one ;)

cc: @DamnClin , @murdos , @pascalgrimaud and @qmonmert

**Pull request merged**